extensions/arrow/src/main/java/io/deephaven/extensions/arrow/ArrowWrapperTools.java

```diff
@@ -0,0 +1,340 @@
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowFileReader;
+import org.apache.arrow.vector.ipc.message.ArrowBlock;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * <pre>
+ * This class contains tools for dealing apache arrow data.
+ * {@link ArrowBooleanColumnSource} - uses {@link BitVector} under the hood, returns Boolean
+ * {@link ArrowByteColumnSource} - uses {@link TinyIntVector} under the hood, returns byte
+ * {@link ArrowCharColumnSource} - uses {@link UInt2Vector} under the hood, returns long
+ * {@link ArrowFloatColumnSource} - uses {@link Float4Vector} under the hood, returns float
+ * {@link ArrowDoubleColumnSource} - uses {@link Float8Vector} under the hood, returns double
+ * {@link ArrowShortColumnSource} - uses {@link SmallIntVector} under the hood, returns short
+ * {@link ArrowIntColumnSource} - uses {@link IntVector} under the hood, returns int
+ * {@link ArrowLongColumnSource} - uses {@link BigIntVector} under the hood, returns long
+ * {@link ArrowUInt1ColumnSource} - uses {@link UInt1Vector} under the hood, returns short
+ * {@link ArrowUInt4ColumnSource} - uses {@link UInt4Vector} under the hood, returns long
+ * {@link ArrowObjectColumnSource<BigInteger>} - uses {@link UInt8Vector} under the hood, returns BigInteger
+ * {@link ArrowLocalTimeColumnSource} - uses {@link TimeMilliVector} under the hood, returns LocalTime
+ * {@link ArrowDateTimeColumnSource} - uses {@link TimeStampVector} under the hood, returns DateTime
+ * {@link ArrowStringColumnSource} - uses {@link VarCharVector} under the hood, returns String
+ * {@link ArrowObjectColumnSource<byte[]>} - uses {@link FixedSizeBinaryVector} under the hood, returns byte[]
+ * {@link ArrowObjectColumnSource<byte[]>} - uses {@link VarBinaryVector} under the hood, returns byte[]
+ * {@link ArrowObjectColumnSource<BigDecimal>} - uses {@link DecimalVector} under the hood, returns BigDecimal
+ * {@link ArrowObjectColumnSource<BigDecimal>} - uses {@link Decimal256Vector} under the hood, returns BigDecimal
+ * {@link ArrowObjectColumnSource<LocalDateTime>} - uses {@link DateMilliVector} under the hood, returns LocalDateTime
+ * </pre>
+ */
+public class ArrowWrapperTools {
+    private static final BufferAllocator rootAllocator = new RootAllocator();
+
+    /**
+     * Reads arrow data from file and returns a query table
+     */
+    public static QueryTable readArrow(final @NotNull String path) {
+        final Helper helper = new Helper(path, rootAllocator);
+        final Shareable context = helper.newShareable();
+        final ArrowFileReader reader = context.getReader();
+        try {
+            int biggestBlock = 0;
+            final List<ArrowBlock> recordBlocks = reader.getRecordBlocks();
+            final int[] blocks = new int[recordBlocks.size()];
```

Comment:
You told me blocks can be bigger than Integer.MAX_VALUE, because they are not stored in arrays. Maybe you are lying, or maybe the code is wrong, or maybe the Java arrow impl can't handle it?

Reply:
It appears that while the initial portion of the parsing allows for `long` sized `RecordBatch`es, that the actual `Vector` implementations that interact with underlying `ArrowBuf` expect `int` based indexing. This is consistent with "best-practices" that arrow implementations limit record batch lengths to MAX_INT.